pkg/controller/deployment/deployment_controller.go

```diff
@@ -518,3 +567,45 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
+
+func selectorUpdatedBefore(d1, d2 *extensions.Deployment) bool {
+	t1, t2 := lastSelectorUpdate(d1), lastSelectorUpdate(d2)
+	return t1.Before(t2)
+}
+
+func lastSelectorUpdate(d *extensions.Deployment) unversioned.Time {
+	t := d.Annotations[util.SelectorUpdateAnnotation]
+	if len(t) > 0 {
+		parsedTime, err := time.Parse(t, time.RFC3339)
+		// If failed to parse the time, use creation timestamp instead
+		if err != nil {
+			return d.CreationTimestamp
+		}
+		return unversioned.Time{Time: parsedTime}
+	}
+	// If it's never updated, use creation timestamp instead
+	return d.CreationTimestamp
+}
+
+// byCreationTimestamp sorts a list of deployments by creation timestamp, using their names as a tie breaker.
+type byCreationTimestamp []extensions.Deployment
+
+func (o byCreationTimestamp) Len() int      { return len(o) }
+func (o byCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o byCreationTimestamp) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(o[j].CreationTimestamp) {
```

Comment:
You probably also want to take into account the `selector-updated-at` annotation too?